kube-client/src/api/subresource.rs

```diff
@@ -248,8 +247,28 @@ where
         self.client.request_text(req).await
     }
 
-    /// Fetch logs as a stream of bytes
-    pub async fn log_stream(&self, name: &str, lp: &LogParams) -> Result<impl Stream<Item = Result<Bytes>>> {
+    /// Fetch logs in the form of an async buf reader
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// # async fn wrapper() -> Result<(), Box<dyn std::error::Error>> {
+    /// # use k8s_openapi::api::core::v1::Pod;
+    /// # use kube::{api::{Api, LogParams}, Client};
+    /// # use futures::{AsyncBufReadExt, TryStreamExt};
+    /// # let client: Client = todo!();
+    /// let pods: Api<Pod> = Api::default_namespaced(client);
+    /// let mut logs = pods
+    ///     .log_stream("my-pod", &LogParams::default()).await?
+    ///     .lines();
+    ///
+    /// while let Some(line) = logs.try_next().await? {
+    ///     println!("{}", line);
+    /// }
+    /// # Ok(())
+    /// # }
+    /// ```
```

Comment:
My attempt for the documentation:

    /// Stream the logs via [`AsyncBufRead`].
    ///
    /// Log stream can be processsed using [`AsyncReadExt`](futures::AsyncReadExt) 
    /// and [`AsyncBufReadExt`](futures::AsyncBufReadExt).
    ///
    /// # Example
    ///
    /// ```no_run
    /// # async fn wrapper() -> Result<(), Box<dyn std::error::Error>> {
    /// # use k8s_openapi::api::core::v1::Pod;
    /// # use kube::{api::{Api, LogParams}, Client};
    /// # let client: Client = todo!();
    /// use futures::{AsyncBufReadExt, TryStreamExt};
    ///
    /// let pods: Api<Pod> = Api::default_namespaced(client);
    /// let mut logs = pods
    ///     .log_stream("my-pod", &LogParams::default()).await?
    ///     .lines();
    ///
    /// while let Some(line) = logs.try_next().await? {
    ///     println!("{}", line);
    /// }
    /// # Ok(())
    /// # }
    /// ```

I think it makes sense to link to the `Async*Ext` traits, since this is what the user will need to work with the response, even more here than on `request_text_stream`.

I am showing the  `futures` import in the example because those are non obvious imports and relevant for the example to work.